scipy/spatial/transform/tests/test_rotation.py

```diff
@@ -1727,3 +1727,178 @@ def test_len_and_bool():
     assert rotation_multi_one
     assert rotation_multi
     assert rotation_single
+
+
+def test_from_davenport_single_rotation():
+    axis = [0, 0, 1]
+    quat = Rotation.from_davenport(axis, 'extrinsic', 90, 
+                                   degrees=True).as_quat()
+    expected_quat = np.array([0, 0, 1, 1]) / np.sqrt(2)
+    assert_allclose(quat, expected_quat)
+
+
+def test_davenport_axes_not_orthogonal():
+    ez = [0, 0, 1]
+    ey = [0, 1, 0]
+    ezy = [0, 1, 1]
+    Rotation.from_davenport([ez, ey], 'e', [0, 0])
+    Rotation.from_davenport([ez, ey, ez], 'e', [0, 0, 0])
```

Comment:
I'm not sure that such semi-tests are customary practice (orthogonal axes are tested extensively in all other places).

I'd keep only part with `pytest.raises`.

Reply:
The idea here is to test that both using only two and only three axes as inputs work, in order to test the handling of multiple axes

Reply:
I think it's better to dedicate a separate test for that, with result checking.

Reply:
Separated into two tests, one for the number of axes and another one for different invalid inputs